tests/ConfigurationTest.php

```diff
@@ -28,6 +28,7 @@ public function tearDown()
     public function test_configuration_directory_is_created_if_it_doesnt_exist()
     {
         $files = Mockery::mock(Filesystem::class.'[ensureDirExists,isDir]');
+        $files->shouldReceive('ensureDirExists')->once()->with(preg_replace('~/valet$~', '', VALET_HOME_PATH), user());
```

Comment:
I wish we could test this--I almost wonder whether we should hard code the path, even if that means having to re-write this test if it ever changed

Reply:
Ya, I was trying to remain consistent between the code and the test, as well as with the code on the next line.
A str_replace would be more readable here too.

Reply:
yah, i think the bummer here is that we're missing a test that the preg_replace is functioning the way we want. I think if I were writing it I would hard code the string here and let this be a test that just breaks later if we change the directory, and then we can just change this test later if we do that. :) Thoughts?

Reply:
The downside to hard-coding it here is that here's how the constant is derived:

https://github.com/laravel/valet/blob/25d6e47eb1506f7763d2acde5705ca07d3d56836/cli/includes/helpers.php#L13

I guess it's shorter if I just extract that out. 

But my thinking has been that then it's very disconnected from the more important next line.

Reply:
Ahhhh ok. That makes a bit sad but I get it . :) Thanks!